src/Cache/MongoLock.php

```diff
@@ -116,19 +127,14 @@ protected function getCurrentOwner(): ?string
         return $this->collection->findOne(
             [
                 '_id' => $this->name,
-                'expiration' => ['$gte' => $this->currentTime()],
+                'expiration' => ['$gte' => $this->getUTCDateTime()],
             ],
             ['projection' => ['owner' => 1]],
         )['owner'] ?? null;
     }
 
-    /**
-     * Get the UNIX timestamp indicating when the lock should expire.
-     */
-    private function expiresAt(): int
+    private function getUTCDateTime(int $additionalSeconds = 0): UTCDateTime
     {
-        $lockTimeout = $this->seconds > 0 ? $this->seconds : $this->defaultTimeoutInSeconds;
-
-        return $this->currentTime() + $lockTimeout;
+        return new UTCDateTime(Carbon::now()->addSeconds($additionalSeconds));
```

Comment:
What allows UTCDateTime to be constructed from a Carbon instance? Is this relying on Carbon's string representation?

I looked at the [Addition and Subtraction](https://carbon.nesbot.com/docs/#api-addsub) docs for Carbon and the strings in the examples don't look correct.

Reply:
The [`Carbon` class extends `DateTime`](https://github.com/briannesbitt/Carbon/blob/master/src/Carbon/Carbon.php#L836), and the `UTCDateTime` [constructor](https://www.php.net/manual/en/mongodb-bson-utcdatetime.construct.php) accepts any `DateTimeInterface`. There is no string transformation.

Reply:
Ah, I missed Carbon's inheritance. SGTM!